docs/upgrade-guide/upgrading-to-0.6.md

```diff
@@ -0,0 +1,177 @@
+# Upgrading to CDKTF version 0.6
+
+CDK for Terraform version 0.6 uses [`constructs`](https://github.com/aws/constructs) version 10. For most users upgrading the versions should be seamless. If you are using more advanced APIs from the `constructs` library, see the [Constructs API Changes](#constructs-api-changes) section.
+
+## Background
+
+Since version 0.6 the `cdktf` package depends on `constructs` v10 (formerly was v3). While that jump might look big, v10 is actually going to be the next major version after v3. The reason for this jump lies in the underlying intention: The `constructs` lib v10 is supposed to stay around for a while and is supposed to offer a very stable API.  
+To achieve this stability in the core functionality a lot has been removed from v10. For further background information you can read into the [PR for v10](https://github.com/aws/constructs/pull/263) or the AWS CDK RFC [Removal of Construct Compatibility Layer](https://github.com/aws/aws-cdk-rfcs/blob/master/text/0192-remove-constructs-compat.md).
+
+While using the v0.6 `cdktf-cli` to deploy stacks that are using an older version of `cdktf` and `constructs` should still work, we recommend upgrading the `cdktf` and `constructs` packages.
+
+## Pre-built providers
+
+All upcoming releases of the [pre-built providers](https://cdk.tf/prebuilt-providers) will depend on `constructs` v10. So we recommend to upgrade your project.
+If you cannot upgrade the `cdktf` version, you can switch to [generating provider bindings](https://github.com/hashicorp/terraform-cdk/blob/main/docs/working-with-cdk-for-terraform/using-providers.md#importing-providers-and-modules) via `cdktf get`.
+
+## Upgrading a TypeScript / JavaScript CDKTF project
+
+To upgrade a TypeScript or JavaScript CDKTF project, adjust the versions in the `package.json` file of your project.
+
+```jsonc
+{
+  // ...
+  "dependencies": {
+    "cdktf": "^0.6.0",
+    "constructs": "^10.0.5"
```

Comment:
Do we warn users if they run 0.6 using an older version of the constructs package?

Reply:
The peer dependency is set to v10, so e.g. for TS/JS there will be a warning from npm / yarn. For the other languages I don't know this but it could have similar effects.
However, we could add a check to the `cdktf` package which could then fail and print an useful error message 👍
Going to leave this discussion open until I added such a check.